addons/gdshell/commands/default_commands/ls.gd

```diff
@@ -0,0 +1,119 @@
+extends GDShellCommand
+
+
+signal done_with_instance(instance: Node)
+
+
+var created_an_instance:= false
```

Comment:
If the node is created in `get_starting_node()` it is not added to the `SceneTree` so we can just later check if the `root_node` is outside the tree and then delete it, if it is.
If the node is not created, it must have come from the `SceneTree` and we want to keep it.
```suggestion
```